internal/flink/command_statement_list.go

```diff
@@ -35,13 +55,31 @@ func (c *command) statementList(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	status, err := cmd.Flags().GetString("status")
+	if err != nil {
+		return err
+	}
+	status = strings.ToUpper(status)
+
+	if status != "" && !slices.Contains(allowedStatuses, status) {
+		return errors.NewErrorWithSuggestions(
+			"invalid value for flag --status",
+			fmt.Sprintf("Please select a value from the following: [%s]", strings.Join(allowedStatuses, ", ")),
+		)
+	}
+
 	statements, err := client.ListAllStatements(environmentId, c.Context.GetCurrentOrganization(), c.Context.GetCurrentFlinkComputePool())
 	if err != nil {
 		return err
 	}
 
 	list := output.NewList(cmd)
 	for _, statement := range statements {
+		// filter out statements with a non-matching status
+		if status != "" && statement.Status.GetPhase() != status {
+			continue
```

Comment:
`continue` can be harder to maintain, should flip the condition and ad only when you need to

Reply:
or just use lo.Filter